src/dataflow/src/sink/kafka.rs

```diff
@@ -190,8 +200,18 @@ where
             );
 
             let ret = move |input: &mut FrontieredInputHandle<_, _, _>| {
+                if shutdown.load(Ordering::SeqCst) {
+                    error!(
+                        "encountered irrecoverable error. shutting down sink: {}",
+                        name
+                    );
+                    *producer.borrow_mut() = None;
+                    return false;
+                }
+
                 let producer: &RefCell<_> = producer.borrow();
                 let producer = &*producer.borrow();
```

Comment:
I know this was like this before, but looking at it again I got really confused, and I think there's something funky with the fact that `std::borrow::Borrow` is in scope and therefore shadowing the `borrow` method on the RefCell. I think it's a bit clearer if you apply this diff?

```diff
diff --git a/src/dataflow/src/sink/kafka.rs b/src/dataflow/src/sink/kafka.rs
index 4adad977..feb8247b 100644
--- a/src/dataflow/src/sink/kafka.rs
+++ b/src/dataflow/src/sink/kafka.rs
@@ -7,7 +7,6 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use std::borrow::Borrow;
 use std::cell::RefCell;
 use std::collections::VecDeque;
 use std::rc::Rc;
@@ -209,10 +208,8 @@ where
                     return false;
                 }
 
-                let producer: &RefCell<_> = producer.borrow();
-                let producer = &*producer.borrow();
-
-                let producer = match producer {
+                let producer = producer.borrow();
+                let producer = match &*producer {
                     Some(producer) => producer,
                     None => return false,
                 };

```